Packs/Workday/Integrations/WorkdayIAMEventsGenerator/WorkdayIAMEventsGenerator.py

```diff
@@ -970,81 +827,158 @@
+                "Org_Level_1": "All R&D",
+                "Org_Level_3": "FWaaP",
+                "Country_Name": "United States Of America",
+                "Org_Level_2": "Enterprise R&D",
+                "Emp_ID": "115074",
+                "Job_Family": "Product Management",
+                "Preferred_Name_-_First_Name": "Chester",
+                "Nikesh Arora": "False",
+                "Management_Level_1": "Nikesh Arora",
+                "Work_Country_Abbrev": "US",
+                "Management_Level_2": "Timmy Turner",
+                "Email_Address": user_email,
+                "Title": "Product Line Manager",
+                "City": "Santa Clara",
+                "Work_State_US_Only": "California",
+                "Job_Code": "5225",
+                "PAN_CF_Okta_Location_Region": "Americas",
+                "Last_Name": last_name,
+                "Job_Function": "Product Management Function",
+                "State": "California",
+                "Exec_Admin_Flag": "N",
+                "Preferred_Name": "Chester McBadbat",
+                "Regular_Employee_Flag": "Y",
+                "Preferred_Name_-_Last_Name": "McBadbat",
+                "Cost_Center_Code": "651116",
+                "Location": "Office - USA - CA - Headquarters"
+            }
+        ]
+    }
+    new_report = report_entry['Report_Entry'][0]
+    for report in integration_context['Report_Entry']:
+        email_address = report.get('Email_Address')
+        if user_email == email_address:
+            return_error(f'User "{user_email}" already exist. Please try another user email.')
+
+    integration_context['Report_Entry'].append(new_report)
+    set_integration_context(integration_context)
 
 
 def get_terminate_report(user_email):
-    if not user_email:
-        set_integration_context(TERMINATE_USER_REPORT)
-    else:
-        integration_context = get_integration_context()
-        reports = TERMINATE_USER_REPORT['Report_Entry']
-        for report in reports:
-            email = report.get('Email_Address')
-            if email == user_email:
-                users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
-                users.append(report)
-                integration_context['Report_Entry'] = users
-                set_integration_context(integration_context)
+    integration_context = get_integration_context()
+    reports = TERMINATE_USER_REPORT['Report_Entry']
+    for report in reports:
+        email = report.get('Email_Address')
+        if email == user_email:
+            users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
+            users.append(report)
+            integration_context['Report_Entry'] = users
+            set_integration_context(integration_context)
+            return
+        else:
+            return_error(f'The user "{user_email}" does not exist. Please choose one of the following: '
```

Comment:
if you have 100 users, and the user you are looking for is at place 50, then the loop will fail at the first user, because  it will go to else branch